source/adapters/hip/memory.hpp

```diff
@@ -502,3 +502,22 @@ struct ur_mem_handle_t_ {
     }
   }
 };
+
+constexpr size_t imageElementByteSize(hipArray_Format ArrayFormat) {
```

Comment:
Does this need to be a fee floating function? Maybe we could hide it in the cpp file behind an anonymous namespace?

Reply:
This function is used by two separate `.cpp` files, and used to be defined 3 times (with different names) in those two files, so I figured it'd be better to have a single definition. If that's not ideal because the function is now available in the entire adapter, what would be the preffered approach? Have two separate definitions of the function in anonymous namespace, or maybe a function that's defined in one `.cpp` file and `extern`ed in another?

Reply:
My bad, I only checked the diffs and it looked like it was only used in `memory.cpp`. I'd vote on one def + `extern`.

Reply:
Single def + extern in latest version